consent-ui/src/app/ais/common/cookie-renewal/CookieRenewalService.ts

```diff
@@ -0,0 +1,57 @@
+import { Injectable } from '@angular/core';
+import { SimpleTimer } from 'ng2-simple-timer';
+import { PsuAuthenticationService } from '../../../api-auth';
+import * as uuid from 'uuid';
+import { ApiHeaders } from '../../../api/api.headers';
+import { SessionService } from '../../../common/session.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class CookieRenewalService {
+  public static TIMER_NAME = 'cookie-renewal-timer';
+  public static ROUTE = 'login';
+
+  constructor(
+    private simpleTimer: SimpleTimer,
+    private psuAuthService: PsuAuthenticationService,
+    private sessionService: SessionService
+  ) {}
+
+  activate(authid): void {
+    const timer = this.getTimer(authid);
+    console.log(new Date().toLocaleString() + ' activate timer in ' + timer);
+    this.simpleTimer.newTimerCD(CookieRenewalService.TIMER_NAME, timer, timer);
```

Comment:
A concurrency issue
Imagine a user is opening 2 authorizations with authId's: 1234, 6789 and cookie lifetime 1 minute:
1. 0 seconds: open auth 1234 - activate(1234) - creates timer TIMER_NAME
2. 57 seconds:  cookieRenewal is about to be called for authId 1234 (1 second till TIMER_NAME)
3. 57 seconds: open auth 6789 - activate(6789) - creates timer TIMER_NAME that will fire 58 seconds from now also timer for 1234 is lost
4. 61 seconds - cookie for authId has **expired**, TIMER_NAME will fire at (57+58)=115 seconds

Reply:
fixed, timername now  contains authid.